rfp-proposals/rfp-proposal-storage-helper.md

```diff
@@ -0,0 +1,142 @@
+# RFP Proposal: `Storage Helper Libraries`
+
+**Name of Project: Storage Helper Libraries
+
+**Link to RFP:** [Storage Helper Libraries](https://github.com/filecoin-project/devgrants/blob/master/rfps/rfp-storage-helper-libraries.md)
+
+**RFP Category:** `devtools-libraries`
+
+**Proposer:** [@deaswang](https://github.com/deaswang)
+
+**Do you agree to open source all work you do on behalf of this RFP and dual-license under MIT and APACHE2 licenses?:** "Yes"
+
+# Project Description
+
+go-filecoin is to handle operations at the detailed, protocol level. Storage Helper Libraries separate from the go-filecoin node but generally reusable across a variety of storage use cases and implementations.
+
+Storage Helper Libraries should include:
+
+- **Data Structure**: file or folder information for DAG
+
+```
+MetaData {
+  path string // file path and name.
+  chunkNum int // the number of chunk.
+  size int //the size of file.
+  cids []string // all chunk cid order by seq.
+}
+```
+
+if store folder, it will be one list of MetaData.
+
+- **File ingestion and splitting**: prepares files for storage deals in dag-compatible way. recurse through folders. split files that are larger than selected filecoin sector size into chunks according to dag.
+
+```
+AddPiece(file): add file to the dag. return one MetaData object.
```

Comment:
Is this DAG stored locally? Can you describe more what this function does?

Reply:
like `go-filecoin client import ./hello.txt`.